### ISSUE TITLE
High CPU load on VSCode extension host after vscode windows closed

### ISSUE THREAD

**Issue by soimy** [BowlerHatLLC/vscode-nextgenas#143]
Platform: Win10
NextgenAS version: 0.6.0
VSCode version: 1.17.1

How to reproduce:
Open some AS3 project with NextgenAS, then close it.
There will be another vscode proccess alive with very high CPU load and very high memory consumption (extension host)
![extension host](https://user-images.githubusercontent.com/1550237/31607415-cf0a9394-b29e-11e7-8b45-ea50a7a21a88.png)

**Comment by soimy** [comment]
After disabling NextgenAS extension, everything fine.

**Comment by humble-ahitofel** [comment]
Hi, this happens to me also: When I close a VSCode editor with AS3 enabled project, it still keeps running consuming CPU and memory. The memory usage caps at somewhere over 2000MB. 
Though this hasn't stopped me using NextGenAS,  when this happens I also end the rogue VSCode processes from my task manager. This does not occur when I open and close a different type project, like a TypeScript project or similar web project. 

![taskmgr_2017-10-18_10-24-16](https://user-images.githubusercontent.com/5386595/31706403-85b80858-b3f1-11e7-9612-d9f1749944a3.png)

Here's some of my specs:

Platform: Microsoft Windows 10 (10.0) Professional 64-bit (Build 15063)
NextGen ActionScript: 0.6.0
VSCode: 1.17.1 (x64 version)

By the way, I have already fully moved to use this as my main development platform for my AIR projects. I think this plugin is very awesome and I'm looking forward for it to progress more.

**Comment by joshtynjala** [comment]
Thanks to both of you. I can reproduce on Windows 10. I think I've found a solution, and I'll test it out for a few days to be sure everything remains stable.

**Comment by piotrzarzycki21** [comment]
I did discover something it maybe a bit related, so I have decided describe it here.

I have opened Moonshine and one project (Flex) - I did couple of changes and close it. I've opened Moonshine once again, but earlier noticed that previous "Java Platform SE binary" process wasn't closed. 

![image](https://user-images.githubusercontent.com/24554795/31895132-a3191cfc-b810-11e7-8821-94f86de46e46.png)

This processes stays open for a long time (sometimes even 5 minutes) once I close IDE. Another observation is that, as long as the old process is open I'm experiencing real slowness of code completion, it's sometimes almost unusable. Once those process got close code completion become more responsive. I did that test several times with same results.

**Comment by joshtynjala** [comment]
It's not related. VSCode and Moonshine both have their own separate code to start and stop the Java process.

Moonshine needs to manually terminate the Java process when it exits because it won't happen automatically. I guess there must be situations where this is not happening.

I've reproduced this issue in Moonshine before, but it's in a situation that a regular user would never encounter. If I'm debugging Moonshine and I manually stop the debugger, Moonshine will be forced to exit, and I think it happens without any kind of event being dispatched in ActionScript. That causes the Java process to remain active. This will only affect developers who are working on Moonshine, and not regular users, though.

When I'm working on Moonshine, I've found a way to avoid this situation so that I don't need to manually kill all of the Java processes. On macOS, I've found that it seems to be safe to go to the Moonshine menu and choose Quit (or use the Command+Q shortcut), and the Java process will exit too. I assume that the File menu -> Exit/Quit on Windows will work the same way. When Moonshine -> Quit is selected from the menu, Moonshine dispatches some kind of event on the global dispatcher that the language server code listens for to manually shutdown the process.
